### PR TITLE
[AWS][AppSec][Medium] Bucket logging not enabled in sample template

### PR DESCRIPTION
**Pull request by junbo75**
Signed-off-by: junbo <junbo@amazon.com>

- Define a new bucket for storing server access logs in the Core stack
- Update the method for referencing resources across stacks to break circular dependency between the Core and Core-Example stacks. Make sure that the Core stack is deployed before the Core-Example stack.
- Verified that CDK can be deployed successfully
- Verified that server access logs can be written to the new bucket after deploy the example stack for AWSCore
- Verified that server access logs can be written to the bucket specified via the context variable after deploy the example stack for AWSMetrics
- Passed automation test.

[aws_core_diff.txt](https://github.com/o3de/o3de/files/6844248/aws_core_diff.txt)

### CODE REVIEW

File: Gems/AWSCore/cdk/example/example_resources_stack.py [hunk 1]
```diff
@@ -42,29 +41,96 @@ def __init__(self, scope: core.Construct, id_: str, project_name: str, feature_n
...
+        user_group = iam.Group.from_group_arn(
+            self,
+            f'{self._project_name}-{self._feature_name}-ImportedUserGroup',
+            core.Fn.import_value(f'{self._project_name}:UserGroup')
+        )
+        admin_group = iam.Group.from_group_arn(
+            self,
+            f'{self._project_name}-{self._feature_name}-ImportedAdminGroup',
+            core.Fn.import_value(f'{self._project_name}:AdminGroup')
+        )
```
Comment (onecent1101) [inline Gems/AWSCore/cdk/example/example_resources_stack.py:56]: As you commented in details, core stack should be deployed before core example stack, where is the change to guarantee it? Or cdk will guarantee it automatically as example s3 creation depends on core stack s3?

Comment (junbo75) [reply Gems/AWSCore/cdk/example/example_resources_stack.py:56]: For the core gem, currently we're using the "cdk deploy --all" command to deploy both stacks (Core and CoreExample). With this changes, users should do the deployment using the exact same command and CDK guarantees that the Core stack is deployed before the CoreExample stack. Deployment will fail if users try to deploy the CoreExample stack without the Core stack.

Comment (junbo75) [reply Gems/AWSCore/cdk/example/example_resources_stack.py:56]: This is also part of the problem I want to get addressed at https://jira.agscollab.com/browse/LYN-5149. We don't have any mechanism to make sure that the Core (Common) stack is deployed before users deploy the CDK application for any specific feature.

Comment (onecent1101) [reply Gems/AWSCore/cdk/example/example_resources_stack.py:56]: actually you can add dependency to stack, stack.add_dependency(stack), like coreexamplestack.add_dependency(corestack)

Comment (junbo75) [reply Gems/AWSCore/cdk/example/example_resources_stack.py:56]: I believe this dependency has already been added automatically if you deploy these two stacks together. Even we have this line added, users can still deploy the example stack using the standard CDK command like below without deploying the core stack:

- cdk deploy CoreExampleStack

File: Gems/AWSCore/cdk/app.py [hunk 2]
```diff
@@ -37,7 +37,7 @@
 
 app = core.App()
 
-core = AWSCore(
+core_construct = AWSCore(
```
Comment (onecent1101) [inline Gems/AWSCore/cdk/app.py:40]: Minor: I believe there was discussion talking about using construct or stack? seems like here the coding standard is inconsistent, core is a construct, but example is a stack.

**Review by onecent1101 — APPROVED** [review]

File: Gems/AWSCore/cdk/app.py (hunk 2 above)
Comment (junbo75) [reply Gems/AWSCore/cdk/app.py:40]: Seems out of scope for this change. We can address this in a separate ticket.

**Review by lmbr-pip — COMMENTED** [review]
LG.

Can you attach cdk diffs to the PR description?

**Comment by junbo75** [comment]
> LG.
> 
> Can you attach cdk diffs to the PR description?

Just uploaded the diffs as an attachment in the description.

**Review by lmbr-pip — APPROVED** [review]
LG

**Pull request merged**